uber/site_sections/registration.py

```diff
@@ -42,20 +42,26 @@ def index(self, session, message='', page='1', search_text='', uploaded_id='', o
 
         attendees = attendees.order(order)
 
-        groups = set()
-        for a in session.query(Attendee) \
-                        .filter(Attendee.first_name == '', Attendee.group_id != None) \
-                        .options(joinedload(Attendee.group)).all():
-            groups.add((a.group.id, a.group.name + (' ({})'.format(a.group.leader.full_name) if a.group.leader else '')))
-
-        if search_text and count == total_count:
-            message = 'No matches found'
-        elif search_text and count == 1 and (not AT_THE_CON or search_text.isdigit()):
-            raise HTTPRedirect('form?id={}&message={}', attendees.one().id, 'This attendee was the only search result')
```

Comment:
This is a much more efficient query than the one it replaces, which should also help, performance-wise.